resources/Microsoft.Windows.Setting.Accessibility/Microsoft.Windows.Setting.Accessibility.psm1

```diff
@@ -584,6 +595,128 @@ class TextCursor
+    static hidden [string] $KeyboardShortcutStatusProperty = 'HotkeyEnabled'
+
+    static [bool] GetStatus()
+    {
+        $FilterStatusArgs = @{  Path = $global:NTAccessibilityRegistryPath; Name = ([ColorFilter]::FilterStatusProperty)}
+        if (-not(DoesRegistryKeyPropertyExist @FilterStatusArgs))
+        {
+            return $false
+        }
+        else
+        {
+            $colorFilterSetting = (Get-ItemProperty @FilterStatusArgs).Configuration
+            return ($colorFilterSetting -eq ([ColorFilter]::FilterStatusValue))
+        }
+    }
+
+    static [int] GetColor()
+    {
+        $FilterColorArgs = @{  Path = $global:ColorFilteringRegistryPath; Name = ([ColorFilter]::FilterColorProperty)}
+        if (-not(DoesRegistryKeyPropertyExist @FilterColorArgs))
+        {
+            return $false
+        }
+        else
+        {
+            $colorFilterSetting = (Get-ItemProperty @FilterColorArgs).FilterType
+            return $colorFilterSetting
+        }        
+    }
+
+    static [nullable[bool]] GetKeyboardShortcutStatus()
+    {
+        $KeyboardShortcutStatusArgs = @{ Path = $global:ColorFilteringRegistryPath; Name = ([ColorFilter]::KeyboardShortcutStatusProperty); }
+        if (-not(DoesRegistryKeyPropertyExist @KeyboardShortcutStatusArgs))
+        {
+            return $false
+        }
+        else
+        {
+            $colorFilterSetting = (Get-ItemProperty @KeyboardShortcutStatusArgs).HotkeyEnabled
+            return $colorFilterSetting
+        }        
+    }
+
+    [ColorFilter] Get()
+    {
+        $currentState = [ColorFilter]::new()
+        $currentState.FilterStatus = [ColorFilter]::GetStatus()
+        $currentState.FilterColor = [ColorFilter]::GetColor()
+        $currentState.KeyboardShortcutStatus = [ColorFilter]::GetKeyboardShortcutStatus()
+        
+        return $currentState
+    }
+
+    [bool] Test()
+    {
+        $currentState = $this.Get()
+        if (($null -ne $this.FilterStatus) -and ($this.FilterStatus -ne $currentState.FilterStatus))
+        {
+            return $false
+        }
+        if ((0 -ne $this.FilterColor) -and ($this.FilterColor -ne $currentState.FilterColor))
+        {
+            return $false
+        }
+        if (($null -ne $this.KeyboardShortcutStatus) -and ($this.KeyboardShortcutStatus -ne $currentState.KeyboardShortcutStatus))
+        {
+            return $false
+        }
+
+        return $true
+    }
+
+    [void] Set()
+    {
+        if (-not $this.Test())
+        {
+            if ($null -ne $this.FilterStatus) 
+            {
+                $FilterStatusArgs = @{ Path = $global:NTAccessibilityRegistryPath; Name = ([ColorFilter]::FilterStatusProperty); }
+                $ColorFilterValue = if ($this.FilterStatus) { ([ColorFilter]::FilterStatusValue) } else { "" }
+                Set-ItemProperty @FilterStatusArgs -Value $ColorFilterValue
+            }
+            
+            if (0 -ne $this.FilterColor) 

```

Comment:
This parameter is an enum, so you should be checking if it is equal to KeepCurrentValue